catalog/plugin/catalog-plugin-security-audit/src/main/java/org/codice/alliance/catalog/plugin/security/audit/SecurityAuditPlugin.java

```diff
@@ -0,0 +1,164 @@
+ */
+public class SecurityAuditPlugin implements AccessPlugin {
+
+    private List<String> auditAttributes = new ArrayList<>();
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest createRequest)
+            throws StopProcessingException {
+        return createRequest;
+    }
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest updateRequest,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        if (updateRequest == null || !isLocal(updateRequest)
+                || MapUtils.isEmpty(existingMetacards)) {
+            return updateRequest;
+        }
+        for (Map.Entry<Serializable, Metacard> entry : updateRequest.getUpdates()) {
+            Metacard updateMetacard = entry.getValue();
+            Metacard existingMetacard = existingMetacards.get(entry.getKey()
+                    .toString());
+            Set<AttributeDescriptor> attributeDescriptors = updateMetacard.getMetacardType()
+                    .getAttributeDescriptors();
+
+            for (AttributeDescriptor descriptor : attributeDescriptors) {
+                String descriptorName = descriptor.getName();
+                if (auditAttributes.contains(descriptorName) && existingMetacard.getAttribute(
+                        descriptorName) != null) {
+                    if (hasAttributeValueChanged(existingMetacard.getAttribute(descriptorName),
+                            updateMetacard.getAttribute(descriptorName))) {
+                        String updateValue;
+                        if (updateMetacard.getAttribute(descriptorName) == null ||
+                                updateMetacard.getAttribute(descriptorName)
+                                        .getValues() == null) {
+                            updateValue = "[DELETED]";
+                        } else {
+                            updateValue = StringUtils.join(updateMetacard.getAttribute(
+                                    descriptorName)
+                                    .getValues(), ",");
+                        }
+                        SecurityLogger.audit(String.format(
+                                "Attribute %s on metacard %s with value(s) %s was updated to value(s) %s",
+                                descriptorName,
+                                updateMetacard.getId(),
+                                StringUtils.join(existingMetacard.getAttribute(descriptorName)
+                                        .getValues(), ","),
+                                updateValue));
+                    }
+                }
+            }
+        }
+
+        return updateRequest;
+    }
+
+    private boolean hasAttributeValueChanged(Attribute oldAttribute, Attribute newAttribute) {
+        // The old attribute is null, so it can't be "updated"
+        if (oldAttribute == null || CollectionUtils.isEmpty(oldAttribute.getValues())) {
```

Comment:
Should this method return `true` if the attribute didn't exist (`oldAttribute == null`) but was added (`newAttribute != null`)?